logger/common.go

```diff
@@ -117,34 +116,35 @@ func NewInfo(containerID string, containerName string, options ...InfoOpt) *dock
 
 // Start starts the actual logger.
 func (l *Logger) Start(uid int, gid int, cleanupTime *time.Duration, ready func() error) error {
```

Comment:
If I understand this new implementation correctly, this method will now return once the Stdout or Stderr `l.sendLogs` returns. Previously, we were waiting for both calls to return. Are there any implications of this change?

Seems like `errgroups` might be a better fit for this use-case: https://godoc.org/golang.org/x/sync/errgroup

Reply:
I tried using `errgroups` actually, but the problem here is we have a `ready()` function, and if using `errgroup`, we are not able to run this function right after we started the two goroutines without waiting for them to finish. And since we have channel here, we do not need wait groups anymore but just wait for results from the channel.

Reply:
I see, makes sense. In that case, what do you think about this? We can pass separate `errChan` objects into `l.sendLogs(l.Stdout)` and `l.sendLogs(l.Stderr)`.

This way, we can leave pretty much everything as-is, but L132 would look something like
```
errGroup.Go(func() error {
   return <-stdoutErrChan
})
errGroup.Go(func() error {
   return <-stderrErrChan
})
err := errGroup.Wait()
```

Reply:
that might be better, will try that!